src/core_plugins/kibana/common/field_formats/types/color.js

```diff
@@ -0,0 +1,53 @@
+import _ from 'lodash';
+import { asPrettyString } from '../../utils/as_pretty_string';
+import { DEFAULT_COLOR } from './color_default';
+import { FieldFormat } from '../field_format';
+
+const convertTemplate = _.template('<span style="<%- style %>"><%- val %></span>');
+
+export class ColorFormat extends FieldFormat {
+  getParamDefaults() {
+    return {
+      fieldType: null, // populated by editor, see controller below
+      colors: [_.cloneDeep(DEFAULT_COLOR)]
+    };
+  }
+
+  findColorRuleForVal(val) {
+    switch (this.param('fieldType')) {
+      case 'string':
+        return _.findLast(this.param('colors'), (colorParam) => {
+          return new RegExp(colorParam.regex).test(val);
+        });
+
+      case 'number':
+        return _.findLast(this.param('colors'), ({ range }) => {
+          if (!range) return;
+          const [start, end] = range.split(':');
+          return val >= Number(start) && val <= Number(end);
+        });
+
+      default:
+        return null;
+    }
+  }
+
+  static id = 'color';
+  static title = 'Color';
+  static fieldType = [
+    'number',
+    'string'
+  ];
+}
+
+ColorFormat.prototype._convert = {
```

Comment:
I tried to move those under the class with [PR - Remove lodash mixin dependencies from Field Formatters](https://github.com/elastic/kibana/pull/12507) but things are messy because  _convert is an object and not a function and the way _convert is used in [content types](https://github.com/nreese/kibana/blob/e655ab3a825aff74f1b9bbfdec14f7b379d7bbe8/src/core_plugins/kibana/common/field_formats/content_types.js) make references to `this` tricky. @Stacey-Gammon and I decided it was best to leave this alone for now.